server/src/main/java/org/elasticsearch/search/searchafter/SearchAfterBuilder.java

```
@@ -154,8 +154,11 @@ private static Object convertValueFromSortType(String fieldName, SortField.Type
         try {
             switch (sortType) {
                 case DOC, INT:
-                    if (value instanceof Number) {
-                        return ((Number) value).intValue();
+                    if (value instanceof Number valueNumber) {
```

Mostly thinking outloud since I'm not entirely sure what a user's expectations would be here.  But when I read this snippet I was wondering if this logic should throw an error instead of defaulting the value to max int.

---

@john-wagster I've added a comment for clarification. 

This is to support the current behaviour for compatibility. Currently, before this PR, all sort on INT fields was treated as Long sort, so search_after of values > Integer.MAX_VALUE were allowed. 
Now to support this and also to support sort on mixed shards of int and long value, we should convert values larger than Integer.MAX_VALUE